packages/nimble-components/src/rich-text-viewer/tests/rich-text-viewer-matrix.stories.ts

```diff
@@ -22,13 +24,100 @@ export default metadata;
 
 // prettier-ignore
 const component = (): ViewTemplate => html`
-    <${richTextViewerTag}></${richTextViewerTag}>
+    <${richTextViewerTag} :markdownValue="${_ => richTextMarkdownString}"></${richTextViewerTag}>
+`;
+
+const viewerSizingTestCase = (
+    [widthLabel, widthStyle]: [string, string],
+    [heightLabel, heightStyle]: [string, string]
+): ViewTemplate => html`
+    <p style="font: 14px/18px Source Sans Pro; margin-bottom: 0px;">${widthLabel}; ${heightLabel}</p>
+    <div style="width: 500px; height: 150px; outline: 1px dotted black">
+        <${richTextViewerTag}
+            style="${widthStyle}; ${heightStyle}"
+            :markdownValue="${_ => richTextMarkdownString}"
+        >
+        </${richTextViewerTag}>
+    </div>
+`;
+
+const viewerDifferentContentTestCase = (
+    [label, markdownContent]: [string, string],
+    [heightLabel, parentHeightStyle]: [string, string]
+): ViewTemplate => html`
+    <p style="font: 14px/18px Source Sans Pro; margin-bottom: 0px;">${label}; ${heightLabel}</p>
+    <div style="width: 300px; outline: 1px dotted black; ${parentHeightStyle}">
+        <${richTextViewerTag}
+            :markdownValue="${_ => markdownContent}"
+        >
+        </${richTextViewerTag}>
+    </div>
+`;
+
+const componentFitToContent = ([widthLabel, widthStyle]: [
+    string,
+    string
+]): ViewTemplate => html`
+    <p style="font: 14px/18px Source Sans Pro; margin-bottom: 0px;">${widthLabel}</p>
+    <${richTextViewerTag} style="${widthStyle}; outline: 1px dotted black" 
```

Comment:
I was a bit confused by these tests initially because the dotted outline doesn't reflect the size of the actual viewer component. For example, in the "width 300px; height 75px" case, the dotted line is taller than the space taken up by the element.

<img width="648" alt="image" src="https://github.com/ni/nimble/assets/10500124/9ab74a55-dddd-4e24-924c-cbaadb9407ec">

I like the idea of these tests but I think they'd be more effective if we could get the dotted line to show the exact size.

(It's possible I missed something, I was looking at this quickly)

Reply:
These tests aim to evaluate the appearance of the component when we set the height and width to it. Also, it shows how the vertical scrollbar is displayed by adjusting the height of both the component and its parent element (dotted outline). 

To indicate the size allocation for each element, I have positioned the size of the parent element on the top right and the size of the component on the left as in the below image.
![image](https://github.com/ni/nimble/assets/123377523/eab7949d-9f78-4226-929f-e34bb44ea8a1)

Could you please let me know if it is okay to have these sizes mentioned in the tests?

Reply:
The challenge is that the Chromatic snapshots are identical for cases that should look different, so the tests aren't actually verifying that the sizing behavior is correct. For example [in the Chrome screenshots](https://www.chromatic.com/test?appId=60e89457a987cf003efc0a5b&id=64b1306799bade1409c030c5) the scrollbar isn't visible so the only indication of the height and width is where the text clips, which is an imperfect indication of its size.

I can see a few options to test this behavior:
1. apply test-only styling to the rich-text-viewer to render its size. e.g. `outline: 1px dashed red`
2. if the final visual design is ready and includes a border, adopt that styling to achieve the same effect
3. change these to be unit tests which call `getComputedStyle()` on the viewer. The downside of this is that it won't test render behaviors at different sizes like word wrapping

I'm fine with any of them but 1 is probably the simplest option to move this PR forward.

Reply:
I have added the test-only outline for the viewer component as suggested in option 1.